plugins/chat/app/services/chat/lookup_channel_threads.rb

```diff
@@ -55,38 +71,17 @@ def can_view_channel(guardian:, channel:, **)
       guardian.can_preview_chat_channel?(channel)
     end
 
-    def fetch_threads(guardian:, channel:, **)
-      read_threads = []
-
-      unread_threads =
+    def fetch_threads(contract:, guardian:, channel:, **)
+      threads =
         threads_query(guardian, channel)
-          .where(<<~SQL)
-            user_chat_thread_memberships_chat_threads.last_read_message_id IS NULL
-              OR tracked_threads_subquery.latest_message_id > user_chat_thread_memberships_chat_threads.last_read_message_id
-          SQL
           .order("tracked_threads_subquery.latest_message_created_at DESC")
-          .limit(MAX_THREADS)
+          .limit(context.limit)
+          .offset(context.offset)
           .to_a
 
-      # We do this to avoid having to query additional threads if the user
-      # already has a lot of unread threads.
-      if unread_threads.length < MAX_THREADS
-        final_limit = MAX_THREADS - unread_threads.length
-        read_threads =
-          threads_query(guardian, channel)
-            .where(<<~SQL)
-              tracked_threads_subquery.latest_message_id <= user_chat_thread_memberships_chat_threads.last_read_message_id
-            SQL
-            .order("tracked_threads_subquery.latest_message_created_at DESC")
-            .limit(final_limit)
-            .to_a
-      end
-
-      threads = unread_threads + read_threads
-
```

Comment:
I explain later why I removed this

Reply:
Yes it is needed so the unread threads always show at the top of the list regardless of how old their last message is, otherwise the following issue occurs:

We are ordering by `latest_message_created_at` , so if you have any unread threads where the `latest_message_created_at ` is older than any threads with no unread messages that have a newer `latest_message_created_at` in the first page of threads, we will not see the unread thread at the top of the list. To reproduce this locally, just clear the `last_read_message_id` of a very old thread that is not on the first page of the thread list for your user. You will see the unread count in the header but that thread will not show at the top of the list until you scroll and load more, then it will jump to the top of the list because of `sortThreads` in the UI.

I could not make a query that loads both the unread and read threads at the same time and order unread at the top, it was too complex and the ordering in SQL would not work consistently with all the includes etc.

Also keep in mind this will probably be simplified by my other PR to use the last_message rather than this subquery.

Reply:
I probably need a better spec for this because this is not apparent at all in the UI if you just e.g. have an unread thread in the first page of threads because the UI also does the sort.